tests/dbgp/dbgpclient.php

```diff
@@ -111,13 +160,69 @@ function runTest( $data, array $commands, array $ini_options = null )
 			$i++;
 		}
 		fclose( $conn );
+		fclose( $ppipes[0] );
+		fclose( $ppipes[1] );
 		proc_close( $php );
 	}
 }
 
-function dbgpRun( $data, $commands, array $ini_options = null)
+class DebugClientIPv6 extends DebugClient
+{
+	protected function getIPAddress()
+	{
+		$records = dns_get_record(php_uname('n'), DNS_AAAA);
+
+		if(!empty($records))
+		{
+			return $records[0]['ipv6'];
+		}
+
+		return "::1";
+	}
+
+	protected function getIP()
+	{
+		return "tcp://[" . $this->getIPAddress() . "]:" . $this->getPort();
+	}
+
+	public static function isSupported()
+	{
+		$ret = true;
+
+		if(!defined("AF_INET6"))
+			return false;
+
+		$socket = socket_create(AF_INET6, SOCK_STREAM, SOL_TCP);
+
+		if($socket === false)
+			return false;
+
+		if($ret && !socket_bind($socket, $this->getIPAddress(), 9990))
+		{
+			$ret = false;
+		}
+
+		if($ret && !socket_listen($socket))
+		{
+			$ret = false;
+		}
+
+		socket_close($socket);
+		unset($socket);
+
+		return $ret;
+	}
+}
+
+function dbgpRun( $data, $commands, array $ini_options = null, $startPort = 9991 )
```

Comment:
Instead of having the start port as an argument, I think I'd rather have that gone - and instead have a flag/constant for IPv4 vs IPv6 - with IPv4 being the default.